### PR TITLE
optional 'set' field in insert permissions, close #216

### LINKED ISSUE #216: alias `x-hasura-user-id` and other `x-hasura-*` variables in mutations
Consider this request payload when creating a todo.

`{"operationName":"insert_todo","variables":{"objects":[{"task":"test-todo5","user_id":"auth0|5b721527280d8b73e2721281","completed":false}]},"query":"mutation insert_todo($objects: [todo_insert_input!]) {\n  insert_todo(objects: $objects) {\n    affected_rows\n    returning {\n      id\n      task\n      completed\n    }\n  }\n}\n"}`

Is there a way to substitute the` user_id` field in a graphql mutation with a Hasura header value on the server side? Ideally with the `X-Hasura-User-Id` header that already has the same information(https://docs.hasura.io/1.0/graphql/manual/auth/index.html).  

I am looking to avoid sending any more authentication/authorization information than I need to. 

This issue comes from @krishkumar
Originally opened [here](https://github.com/wawhal/todo-app-graphql-engine-auth0/issues/2).

### PR DESCRIPTION
**Pull request by rakeshkky**
### Description

What component does this PR affect? 

- [x] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [x] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue

close #216 

### Solution and Design

Additional and optional `set` field in`insert` permisison in which `admin` can enforce input
values for certain columns to be a constant value or a session-variable (`X-Hasura-` header)

Example for modified insert permission API:

```http
POST /v1/query

{
  "type": "create_insert_permission",
  "args": {
    "table": "article",
    "role": "student",
    "permission": {
      "check": { "author_id": "X-Hasura-Student-Id" },
      "set": { "author_id": "X-Hasura-Student-Id", "is_published": false },
      "allow_upsert": true
    }
  }
}
```

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [x] I have added required tests.

### CODE REVIEW

File: server/src-lib/Hasura/GraphQL/Resolve/Mutation.hs [hunk 1]
```diff
@@ -61,13 +61,15 @@ convertMutResp qt ty selSet =
...
   -> m [(PGCol, S.SQLExp)]
-convertRowObj val =
-  flip withObject val $ \_ obj -> forM (Map.toList obj) $ \(k, v) -> do
+convertRowObj setVals val =
+  flip withObject val $ \_ obj -> do
+  inpVals <- forM (Map.toList obj) $ \(k, v) -> do
     prepExpM <- asPGColValM v >>= mapM prepare
     let prepExp = fromMaybe (S.SEUnsafe "NULL") prepExpM
     return (PGCol $ G.unName k, prepExp)
+  return $ Map.toList setVals <> inpVals
```
[the remark this answers is not in the document]
Comment (rakeshkky) [reply server/src-lib/Hasura/GraphQL/Resolve/Mutation.hs]: Not generating an input field for columns that are specified in `set` permission in `table_insert_input`. Hence `setVals` and `inpVals` are always mutually exclusive.

**Comment by shahidhk** [comment]
Needs console changes.

**Comment by coco98** [comment]
Let’s merge console changes on this PR too? That way we can all test and
validate.

**Comment by shahidhk** [comment]
@coco98 Just created a separate issue. I'll close it then?

**Comment by coco98** [comment]
No need to close right? We can just have this PR span multiple issues.
Also, do we have python tests for this PR?

**Comment by shahidhk** [comment]
@coco98 Cool.
Yes, tests are added for the API. You can find it in the files changed.

**Comment by shahidhk** [comment]
@rakeshkky Can you resolve conflicts for server/

File: console/src/components/Services/Data/TablePermissions/Permissions.js [hunk 2]
```diff
@@ -93,6 +216,56 @@ class Permissions extends Component {
       console.log(e);
```
Comment (praveenweb) [inline console/src/components/Services/Data/TablePermissions/Permissions.js]: Remove console.log

File: console/src/components/Services/Data/TablePermissions/Permissions.js [hunk 3]
```diff
@@ -93,6 +216,56 @@ class Permissions extends Component {
...
+    let showInsertPrefix = false;
+    try {
+      showInsertPrefix = semverCheck('insertPrefix', version);
+      if (showInsertPrefix) {
+        this.setState({ ...this.state, showInsertPrefix: true });
+      } else {
+        this.setState({ ...this.state, showInsertPrefix: false });
+      }
+    } catch (e) {
+      console.log(e);
```
Comment (praveenweb) [inline console/src/components/Services/Data/TablePermissions/Permissions.js]: Remove console.log

**Pull request merged**